src/sentry/dynamic_sampling/tasks/custom_rule_notifications.py

```diff
@@ -30,8 +30,8 @@
     queue="dynamicsampling",
     default_retry_delay=5,
     max_retries=5,
-    soft_time_limit=2 * 60 * 60,  # 2hours
-    time_limit=2 * 60 * 60 + 5,
+    soft_time_limit=1 * 60,  # 1 minute
```

Comment:
Should this be greater than `MAX_TASK_SECONDS` (7 mins)? Or should we lower the `MAX_TASK_SECONDS`?

I see that in this task we use `TimedIterator` which I guess is connected to the `MAX_TASK_SECONDS` value

Reply:
The time limits of the `TimedIterator` were chosen empirically based on the query performance observed in production. It’s slightly longer than the query runtime.

One thing to be careful about is how query performance could degrade with additional customers (since it scans multiple organizations in batches). I recommend setting the value to a few minutes longer than the duration it takes for each task to run.